# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/valgrind/hotfix-ppc32-syscalls.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

Debugging Firefox:

--25422-- WARNING: unhandled ppc32-linux syscall: 341
--23768-- WARNING: unhandled ppc32-linux syscall: 342
--25450-- WARNING: unhandled ppc32-linux syscall: 333
 
--- valgrind-3.25.1/coregrind/m_syswrap/syswrap-ppc32-linux.c.vanilla	2025-08-30 13:32:14.366264941 +0200
+++ valgrind-3.25.1/coregrind/m_syswrap/syswrap-ppc32-linux.c	2025-08-30 13:35:43.436354092 +0200
@@ -998,7 +998,7 @@
    LINXY(__NR_accept,            sys_accept),           // 330
    LINXY(__NR_getsockname,       sys_getsockname),      // 331
    LINXY(__NR_getpeername,       sys_getpeername),      // 332
-
+   LINXY(__NR_socketpair,        sys_socketpair),       // 333
    LINX_(__NR_send,              sys_send),             // 334
    LINX_(__NR_sendto,            sys_sendto),           // 335
    LINXY(__NR_recv,              sys_recv),             // 336
@@ -1006,6 +1006,8 @@
    LINX_(__NR_shutdown,          sys_shutdown),         // 338
    LINX_(__NR_setsockopt,        sys_setsockopt),       // 339
 
+   LINX_(__NR_sendmsg,           sys_sendmsg),          // 341
+   LINXY(__NR_recvmsg,           sys_recvmsg),          // 342
    LINXY(__NR_recvmmsg,          sys_recvmmsg),         // 343
    LINXY(__NR_accept4,           sys_accept4),          // 344
 
